### PR TITLE
Improvements and bug fixes for the dashboard view page

### PR DESCRIPTION
**Pull request by this**
## Purpose
Improvements and bug fixes for the dashboard view page.

- Asynchronous widget loading.
  - Widgets are fetched, loaded, and rendered asynchronously which improves overall performance and UX.
  - Fetching/loading status is displayed with the message and a progress bar.
  - If fetching failed, then user can re-try,
ezgif com-video-to-gif
- Widget caching.
  - Once a widget is successfully fetched and loaded, it will be caches so that next time it will be rendered immediatly.
  - This improves improves overall performance and UX.
ezgif com-video-to-gif 1
- Fixes theme changing issues.
  - Fix issue #894 for the dashboard view page.
ezgif com-video-to-gif 2
ezgif com-video-to-gif 3
- Revamped User menu.
  - Fix issue #879 for the dashboard view page.
image
- Code clean-up 

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? **yes**
 - Ran FindSecurityBugs plugin and verified report? **yes**
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? **yes**

## Test environment
Node v8.8.1, NPM v5.4.2

### CODE REVIEW

File: components/dashboards-web-component/src/auth/SecuredRouter.jsx [hunk 1]
```diff
@@ -102,8 +102,8 @@ export default class SecuredRouter extends Component {
                 <Route path='/designer/:dashboardId/*' component={DashboardDesigner} />
 
                 {/* Dashboard view */}
-                <Route exact path='/dashboards/:id' component={DashboardView} />
-                <Route path='/dashboards/:id/*' component={DashboardView} />
+                <Route exact path='/dashboards/:dashboardId/:pageId?/:subPageId?' component={DashboardViewPage} />
+                {/*<Route path='/dashboards/:id/*' component={DashboardView} />*/}
```
Comment (CharukaK) [inline components/dashboards-web-component/src/auth/SecuredRouter.jsx]: Shall we remove this commented out Route?

File: components/dashboards-web-component/src/common/WidgetRenderer.jsx [hunk 2]
```diff
@@ -0,0 +1,219 @@
...
+                window.eval(response.data);
+                if (this.getWidgetClass()) {
+                    // TODO: 5/5/18 Wire pub/sub
+                    this.updateWidgetLoadingStatus(WidgetLoadingStatus.LOADED);
+                } else {
+                    this.updateWidgetLoadingStatus(WidgetLoadingStatus.FETCHING_FAIL);
+                }
+            })
+            .catch(error => {
+                console.warn(`Fetching widget '${this.widgetName}' failed.` + error.message);
```
Comment (CharukaK) [inline components/dashboards-web-component/src/common/WidgetRenderer.jsx]: Shall we check for the availability of the console object as this is not supported by all browsers? ex: IE<10

File: components/dashboards-web-component/src/common/styles/dashboard-renderer-styles.css [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+.dashboard-container {
+    position: relative;
+    width: 100%;
+    height: calc(100vh - 64px);
+    max-height: 100vh;
+    padding: 0;
+    margin: 0;
+    border: 0;
+}
```
Comment (erangatl) [inline components/dashboards-web-component/src/common/styles/dashboard-renderer-styles.css]: Add file ending line

File: components/dashboards-web-component/src/common/WidgetRenderer.jsx (hunk 2 above)
Comment (this) [reply components/dashboards-web-component/src/common/WidgetRenderer.jsx]: IMO, we don't need to support IE<10.
Let's change this log to debug level

File: components/dashboards-web-component/src/utils/WidgetClassRegistry.js [hunk 4]
```diff
@@ -65,5 +65,25 @@ global.dashboard = {};
 global.dashboard.registerWidget = function (widgetId, widgetObj) {
     console.warn('[DEPRICATED] \'window.dashboard.registerWidget(widgetId, widgetObj)\' function is depricated. '
                  + 'Instead please use \'Widget.registerWidgetClass(widgetName, widgetClass)\' function.');
+    if (_extendsFromDepricatedWidgetClassVersion(widgetId, widgetObj)) {
+        console.warn('[DEPRICATED] \'@wso2-dashboards/widget@1.0.0\' is depricated. Please user a newer version.');
```
Comment (CharukaK) [inline components/dashboards-web-component/src/utils/WidgetClassRegistry.js]: Shall we check for the availability of the console object as it is not available in all browsers? ex: IE<10

Comment (this) [reply components/dashboards-web-component/src/utils/WidgetClassRegistry.js]: IMO, we don't need to support IE<10. However as per https://developer.mozilla.org/en-US/docs/Web/API/Console/warn `console.warn` is supported in IE8.

**Pull request merged**